src/System.IO/src/System/IO/BinaryReader.cs

```diff
@@ -9,6 +9,8 @@ namespace System.IO
 {
     public class BinaryReader : IDisposable
     {
+        private static readonly Encoding NoBomUTF8 = new UTF8Encoding();
```

Comment:
Please revert all of the changes under /System/IO.  These files are only used right now with CoreRT, which means we don't have any test coverage running regularly in CI.